packages/core-mobile/app/new/contexts/SignupProvider.tsx

```diff
@@ -0,0 +1,141 @@
+import { TotpChallenge } from '@cubist-labs/cubesigner-sdk'
+import React, {
+  createContext,
+  Dispatch,
+  ReactNode,
+  SetStateAction,
+  useCallback,
+  useContext,
+  useMemo,
+  useState
+} from 'react'
+import SeedlessService from 'seedless/services/SeedlessService'
+import AnalyticsService from 'services/analytics/AnalyticsService'
+import { useRouter } from 'expo-router'
+import { copyToClipboard } from 'new/utils/clipboard'
+import { TotpErrors } from 'seedless/errors'
+import { Result } from 'types/result'
+import { hideLogoModal, showLogoModal } from 'new/components/LogoModal'
+import { OidcAuth } from 'new/types'
+
+export interface SignupContextState {
+  handleAccountVerified: () => Promise<void>
+  handleBack: () => void
+  handleCopyCode: () => void
+  onVerifyCode: (code: string) => Promise<Result<undefined, TotpErrors>>
+  onVerifySuccess: () => void
+  goToEnterCodeManually: () => void
+  goToScanQrCode: () => void
+  goToVerifyCode: () => void
+  totpKey?: string
+  totpChallenge?: TotpChallenge
+  setTotpChallenge: Dispatch<SetStateAction<TotpChallenge | undefined>>
+  oidcAuth?: OidcAuth
+  setOidcAuth: Dispatch<SetStateAction<OidcAuth | undefined>>
+  allowsUserToAddLater: boolean
+}
+
+export const SignupContext = createContext<SignupContextState>(
+  {} as SignupContextState
+)
+
+export const SignupProvider = ({
+  children
+}: {
+  children: ReactNode
+}): React.JSX.Element => {
+  const router = useRouter()
+  const [oidcAuth, setOidcAuth] = useState<OidcAuth>()
+  const [totpChallenge, setTotpChallenge] = useState<TotpChallenge>()
+
+  const totpKey = useMemo(() => {
+    if (totpChallenge?.totpUrl) {
+      return (
+        new URL(totpChallenge.totpUrl).searchParams.get('secret') ?? undefined
+      )
+    }
+  }, [totpChallenge])
+
+  const handleCopyCode = useCallback((): void => {
+    totpKey && copyToClipboard(totpKey, 'Code copied')
+  }, [totpKey])
+
+  const handleBack = useCallback((): void => {
+    router.canGoBack() && router.back()
+  }, [router])
+
+  const goToEnterCodeManually = (): void => {
+    router.navigate('./copyCode')
+  }
+
+  const goToScanQrCode = (): void => {
+    router.navigate('./scanQrCode')
+  }
+
+  const goToVerifyCode = (): void => {
+    router.push('./verifyCode')
+  }
```

Comment:
I think we can have these routing functions in the screens, even if they might be duplicated in some of them.(they are just one line function call) By keeping the routing function calls only within the screens, I believe that when you look at a screen’s code, you can easily see which other screens it’s connected to at a glance.